### PR TITLE
build: add hardhat for testnet deploys

### PR DESCRIPTION
**Pull request by kerzhner**
[hardhat](https://www.npmjs.com/package/hardhat-deploy) provides a nice set of tools for managing deployments across multiple networks. The downside of this work is that there are now 2 compilation and deploy strategies.
1. For Ganache: we have a custom strategy that uses etherlime.
2. For all other networks, we have hardhat.

In the future, it would be nice to migrate all compilation and deployment to hardhat: https://github.com/statechannels/statechannels/issues/2877

Fixes https://github.com/statechannels/statechannels/issues/2875

### CODE REVIEW

File: packages/nitro-protocol/hardhat.config.ts [hunk 1]
```diff
@@ -19,7 +19,11 @@ module.exports = {
   solidity: '0.6.12',
   namedAccounts: {
     deployer: {
-      rinkeby: 0,
+      // Address used for the initial deploy of the contracts.
+      // Ask @kerzhner for the private key if you would like to deploy from the same address
+      // Otherwise uncomment the line below
+      //rinkeby: 0,
+      rinkeby: '0x87612aAD373586A38062c29F833A2AbC72038591',
```
Comment (andrewgordstewart) [inline packages/nitro-protocol/hardhat.config.ts:26]: I guess this is also a `build: ` commit, since you're not fixing any code?

Comment (kerzhner) [reply packages/nitro-protocol/hardhat.config.ts:26]: `build` is probably a better characterization.

**Review by andrewgordstewart — APPROVED** [review]

File: packages/nitro-protocol/readme.md [hunk 2]
```diff
@@ -101,3 +97,11 @@ See https://docusaurus.io/docs/en/installation for more information.
 NB: you may run into difficulty running `docgen / solidoc` if you have the native solc compiler installed at the incorrect version number. You may refer to the circle `config.yml` at the monorepo root to check which version is being used as a part of our continuous integration.
 
 To add a new version of the docs, follow the instructions at https://docusaurus.io/docs/en/tutorial-version. We try to keep the documentation version in sync with the @statechannels/nitro-protocol npm package.
+
+## Deploying contracts to a testnet
+
+The current [hardhat](https://www.npmjs.com/package/hardhat-deploy) config works with Rinkeby testnet. It can be added on to for any other network.
+
+```
+INFURA_TOKEN=[your token here] RINKEBY_DEPLOYER_PK=[private key used for rinkeby deploy] npx hardhat deploy --network rinkeby
```
Comment (geoknee) [inline packages/nitro-protocol/readme.md:106]: Could we use `yarn` instead of `npx`?

Comment (kerzhner) [reply packages/nitro-protocol/readme.md:106]: Good call! Do you mind making the change in your PR https://github.com/statechannels/statechannels/pull/2893?

**Pull request merged**